src/Raven.Server/Documents/Queries/Results/MapReduceQueryResultRetriever.cs

```diff
@@ -17,6 +17,9 @@ public MapReduceQueryResultRetriever(DocumentDatabase database, IndexQueryServer
             : base(Constants.Documents.Indexing.Fields.ReduceKeyValueFieldName, database, query, queryTimings, documentsStorage, context, fieldsToFetch, includeDocumentsCommand, includeCompareExchangeValuesCommand, includeRevisionsCommand)
         {
         }
+
+        //todo maciej: this should be done in other way
```

Comment:
why?

Reply:
Leftover from the first commit ;-) There is a method for handling getting results for us. It's returning the whole document or projecting the results.